### ISSUE TITLE
Question: Has a text/plain content parser plugin been published?

### ISSUE THREAD

**Issue by lfurzewaddock** [fastify/fastify#1269]
## 💬 Questions and Help

If not, what's best practice? Should we roll our own or try to use express body-parser?

Thanks

**Comment by jsumners** [comment]
You can simply use the built-in parser (https://www.fastify.io/docs/latest/ContentTypeParser/#body-parser):

```js
fastify.addContentTypeParser('text/plain', { parseAs: 'string' }, function (req, body, done) {
  done(null, body)
})
```

**Comment by lfurzewaddock** [comment]
Thanks for the quick response @jsumners. I think it would be worth including your snippet in the docs to avoid people like me looking for a quick fix, asking silly questions. ;-)

**Comment by jsumners** [comment]
PRs for improved documentation are always welcome.

**Comment by mcollina** [comment]
Maybe we should just ship this? I don't remember why we shouldn't.

**Comment by lfurzewaddock** [comment]
I could publish a plugin if that helps? Otherwise I could just add the snippet to the docs, but most people would abstract that away into a plugin anyway.

Its great that everything is low level and can be customised but something like this may give the wrong impression to new users early on during evaluation that it is just harder to use than you would expect, or worse, that it is unfinished and may raise unfounded concerns about other, more advanced aspects of Fastify.

**Comment by mcollina** [comment]
+1 on having a plugin!

**Comment by cemremengu** [comment]
Isn't a plugin somewhat overkill for this? 🤔

**Comment by cemremengu** [comment]
@lfurzewaddock I agree with you so I think that we should bake this into the core for convenience, since it's very basic, instead of a plugin. Not sure if this was discussed before, I personally expect the framework to come with batteries-included for these kind of things.

@fastify/fastify  WDYT ?

**Comment by mcollina** [comment]
I'm +1 on adding this to core.

@delvedor?

**Comment by delvedor** [comment]
Let's add it to core 👍

**Comment by cemremengu** [comment]
@lfurzewaddock would you like to send a PR for this?

**Comment by lfurzewaddock** [comment]
OK @cemremengu, I'll have a look.

**Comment by lfurzewaddock** [comment]
@cemremengu, I'm ready to commit my changes but 7 unit tests are failing in pre-commit. These tests (see below) are related to the http2 module and were failing before I made any changes to source, so I believe they are unrelated. I expect they're failing as I'm on Windows 10, but using bash in WSL (Windows Subsystem for Linux).

Will you accept the PR if I force (--no-verify) the commit? I expect the tests would pass on Travis CI.

```(node:6874) ExperimentalWarning: The http2 module is an experimental API.
test/listen.test.js ................................. 38/45 30s
  listen twice on the same port
  not ok should be equal
    +++ found
    --- wanted
    -[null]
    +"http://127.0.0.1:62677"
    compare: ===
    at:
      line: 146
      column: 9
      file: test/listen.test.js
      function: s2.listen
    stack: |
      s2.listen (test/listen.test.js:146:9)
      Server.wrap (fastify.js:399:9)
    source: |
      t.is(address2, null)

  listen twice on the same port
  not ok expect truthy value
    at:
      line: 147
      column: 9
      file: test/listen.test.js
      function: s2.listen
    stack: |
      s2.listen (test/listen.test.js:147:9)
      Server.wrap (fastify.js:399:9)
    source: |
      t.ok(err2)

  listen twice on the same port callback with (err, address)
  not ok should be equal
    +++ found
    --- wanted
    -[null]
    +"http://127.0.0.1:62678"
    compare: ===
    at:
      line: 163
      column: 9
      file: test/listen.test.js
      function: s2.listen
    stack: |
      s2.listen (test/listen.test.js:163:9)
      Server.wrap (fastify.js:399:9)
    source: |
      t.is(address2, null)

  listen twice on the same port callback with (err, address)
  not ok expect truthy value
    at:
      line: 164
      column: 9
      file: test/listen.test.js
      function: s2.listen
    stack: |
      s2.listen (test/listen.test.js:164:9)
      Server.wrap (fastify.js:399:9)
    source: |
      t.ok(err2)

  listen twice on the same port without callback rejects
  not ok timeout!
    signal: SIGTERM
    handles:
      - type: Server
        events:
          - request
          - connection
          - clientError
        connectionKey: '4:127.0.0.1:0'
      - type: Server
        events:
          - request
          - connection
          - clientError
        connectionKey: '4:127.0.0.1:62711'
    expired: TAP
    stack: |
      emit (node_modules/signal-exit/index.js:77:11)
      process.listener (node_modules/signal-exit/index.js:91:7)
      process.emit (node_modules/source-map-support/source-map-support.js:461:21)
      processEmit (node_modules/signal-exit/index.js:155:32)
    test: listen twice on the same port without callback rejects

  listen twice on the same port without callback rejects with (address)
  not ok test count !== plan
    +++ found
    --- wanted
    -2
    +0
    results:
      ok: false
      count: 0
      pass: 0
      fail: 1
      bailout: false
      todo: 0
      skip: 0
      plan:
        start: 1
        end: 2
        skipAll: false
        skipReason: ''
        comment: ''
      failures:
        - tapError: incorrect number of tests

  not ok test count !== plan
    +++ found
    --- wanted
    -1
    +23
    results:
      ok: false
      count: 23
      pass: 20
      fail: 4
      bailout: false
      todo: 0
      skip: 0
      plan:
        start: null
        end: null
        skipAll: false
        skipReason: ''
        comment: ''
      failures:
        - ok: false
          id: 12
          time: 62.502
          name: listen twice on the same port
        - ok: false
          id: 13
          time: 39.836
          name: 'listen twice on the same port callback with (err, address)'
        - ok: false
          id: 23
          time: 27979.505
          name: listen twice on the same port without callback rejects
        - tapError: no plan
```

**Comment by cemremengu** [comment]
@lfurzewaddock sure, send in the PR for review and we will take a look :)

**Comment by lfurzewaddock** [comment]
@cemremengu, PR #1280 ready

**Comment by cemremengu** [comment]
Landed in #1280
